src/resources/entities/character.py

```diff
@@ -25,34 +23,44 @@ def __init__(self, current_level: Level, symbol: str = "$") -> None:
         )
         self.playing = True
 
-    def start(self) -> None:
-        """Starts the movement controls in a separate thread"""
-        Thread(target=self.keyboard_input, args=()).start()
+    def press(self, key: Key) -> Union[None, bool]:
+        """Reads keyboard input"""
+        try:
+            self.level.board[self.y][self.x] = self.ground_symbol
+            if key.char == 'a':
+                if self.x > 1:
+                    self.x -= 1
+            if key.char == 'd':
+                if self.x < self.level.width - 2:
+                    self.x += 1
+            if key.char == 'w':
+                if self.y > 1:
+                    self.y -= 1
+            if key.char == 's':
+                if self.y < self.level.height - 2:
+                    self.y += 1
+            if key.char == 'p':
+                self.playing = False
+                return False
+            self.ground_symbol = self.level.board[self.y][self.x]
+        except AttributeError:
+            self.playing = False
+
+    def release(self, key: Key) -> bool:
+        """On key release"""
+        try:
+            if key.char not in ('w', 'a', 's', 'd', 'p'):
+                self.playing = False
```

Comment:
We should have a more specific exit condition. Adding new user buttons will require coming here to change this list which will become unmanageable and is also error prone. For now just pick a button as the exit condition and check if the user click it.

Reply:
Originally I had only mapped p to close the game. Any button will do